### PR TITLE
VEN-814 | support "Add New" additional services

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Implemented `AdditionalServicesForm` component for used it for both adding and editing services
- Removed `AdditionalServicesFields` and moved its elements and unit tests to `AdditionalServicesForm`
- Added unit field to the additional services modal

### Closes :no_good_woman:
**[VEN-814:](https://helsinkisolutionoffice.atlassian.net/browse/VEN-814)** FE - Pricing page: support "Add New" additional services

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/f083eeb9dbf3323360dc5a572c283589aeb11671/src/features/pricing/additionalServicePricing/form/__tests__/AdditionalServicesForm.test.tsx#L13

### Manual testing :construction_worker_man:
- Go to the pricing page
- On the "Lisäpalvelut" table, click on "Lisää uusi" button
- Fill in the "Lisää uusi" modal and submit 

## Screenshots :camera_flash:
image
**Adding**
image
**Editing:**
image

### CODE REVIEW

File: src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+};
+
+const OPTIONAL_SERVICES = [
+  ProductServiceType.SUMMER_STORAGE_FOR_DOCKING_EQUIPMENT,
+  ProductServiceType.SUMMER_STORAGE_FOR_TRAILERS,
+  ProductServiceType.PARKING_PERMIT,
+  ProductServiceType.DINGHY_PLACE,
+];
+
+const AdditionalServicesModal = ({
```
Comment (TatuArvela) [inline src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx:69]: This should be called `AdditionalServicesForm`, as the actual Modal is not included

File: src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  priceUnit: PriceUnits | null;
+  taxPercentage: AdditionalProductTaxEnum | null;
+  period?: PeriodType | null;
+}
+
+export interface AdditionalServicesModalProps {
+  initialValues?: AdditionalServiceValues | null;
+  periodOptions: PeriodType[];
+  onSubmit: (values: AdditionalServiceValues) => void;
+  closeModal: () => void;
```
Comment (TatuArvela) [inline src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx:43]: This should probably be named `onCancel`?

**Review by TatuArvela — CHANGES_REQUESTED** [review]
A couple of comments about naming, otherwise LGTM 👍

File: src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx (hunk 1 above)
Comment (Nizar-Rahme) [reply src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx]: Changed.

File: src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx]: Renamed.

Comment (TatuArvela) [reply src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx]: Nitpicking: it's now `onClose`, but I meant `onCancel` in relation to the form buttons, as the action is performed when clicking `Cancel`.

**Review by TatuArvela — APPROVED** [review]
One more comment, but approved!

File: src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx (hunk 2 above)
Comment (Nizar-Rahme) [reply src/features/pricing/additionalServicePricing/modal/AdditionalServicesModal.tsx]: Oops, right! 
Done.

**Pull request merged**